metashare/api/serializers.py

```diff
@@ -237,6 +280,20 @@ def get_pr_url(self, obj) -> Optional[str]:
             return f"https://github.com/{repo_owner}/{repo_name}/pull/{pr_number}"
         return None
 
+    def update(self, instance, validated_data):
+        originating_user_id = getattr(
+            getattr(self.context.get("request"), "user", None), "id", None
+        )
+        if instance.assigned_dev != validated_data["assigned_dev"]:
+            orgs = instance.scratchorg_set.filter(org_type=SCRATCH_ORG_TYPES.Dev)
+            for org in orgs:
+                org.queue_delete(originating_user_id=originating_user_id)
+        if instance.assigned_qa != validated_data["assigned_qa"]:
+            orgs = instance.scratchorg_set.filter(org_type=SCRATCH_ORG_TYPES.QA)
+            for org in orgs:
+                org.queue_delete(originating_user_id=originating_user_id)
```

Comment:
This should work... On the front end, I had been checking if the _new_ `assigned_* != org.owner_gh_username`, so that it would not delete if they somehow got out of sync and then you were re-assigning to the owner. But that's a stretch as far as actual use-cases, so I'm okay if you prefer this.

Reply:
I do somewhat prefer this, yes.